homeassistant/components/knx/sensor.py

```diff
@@ -28,22 +32,34 @@ async def async_setup_platform(
     entities = []
     for device in hass.data[DOMAIN].xknx.devices:
         if isinstance(device, XknxSensor):
-            entities.append(KNXSensor(device))
+            entities.append(KNXSensor(hass, device))
     async_add_entities(entities)
 
 
 class KNXSensor(KnxEntity, Entity):
     """Representation of a KNX sensor."""
 
-    def __init__(self, device: XknxSensor) -> None:
-        """Initialize of a KNX sensor."""
+    def __init__(self, hass: HomeAssistantType, device: XknxSensor):
+        """Initialize a KNX sensor."""
         self._device: XknxSensor
+        if device.ha_value_template is not None:
+            device.ha_value_template.hass = hass
         super().__init__(device)
 
     @property
     def state(self) -> StateType:
         """Return the state of the sensor."""
-        return self._device.resolve_state()
+        state = self._device.resolve_state()
+        template = self._device.ha_value_template
+        if template is not None and state is not None:
+            try:
+                return template.async_render({"value": state})
```

Comment:
mypy: this returns `Any`. Shall this be casted to `StateType` or ignored with `# type: ignore[no-any-return]`?